lib/cluster/workers/IDCardRenewer.ts

```diff
@@ -0,0 +1,115 @@
+import { isMainThread, parentPort } from 'worker_threads';
+import Redis from '../../service/cache/redis';
+import { JSONObject } from 'kuzzle-sdk';
+
+export class IDCardRenewer {
+  private disposed = false;
+  private redis : Redis = null;
+  private refreshTimer : ReturnType<typeof setInterval>;
+  private nodeIdKey : string = null;
+  private refreshDelay = 2000;
+  private parentPort : any = null;
```

Comment:
Why do you need to keep an instance reference on the global `parentPort` variable?

Reply:
Just to be able to mock it in the tests